### PR TITLE
Add and configure Sentry gem

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds the Sentry integration, similarly to what we've done with other applications and the current Sentry documentation.

### Ticket(s)

https://mitlibraries.atlassian.net/browse/TCO-88

### Accessibility

- [ ] ANDI or Wave has been run in accordance to [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened 
      as new issues (link to those issues in the Pull Request details above)
- [x] There are no accessibility implications to this change

### Documentation

- [x] Project documentation has been updated, and yard output previewed
- [ ] No documentation changes are needed

### ENV

- [x] All new ENV is documented in README.
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod.
- [ ] ENV has not changed.

### Stakeholders

- [ ] Stakeholder approval has been confirmed
- [x] Stakeholder approval is not needed

### Dependencies and migrations

YES dependencies are updated

NO migrations are included

### Code

- [ ] I have confirmed that the code works as intended.
- [ ] Any CodeClimate issues have been fixed or confirmed as
added technical debt.

### Documentation

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message).
- [ ] The documentation has been updated or is unnecessary.
- [ ] New dependencies are appropriate or there were no changes.

### Testing

- [ ] There are appropriate tests covering any new functionality.
- [ ] No additional test coverage is required.

### CODE REVIEW

File: config/initializers/sentry.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+Sentry.init do |config|
+  return unless ENV.has_key?('SENTRY_DSN')
+  config.dsn = ENV.fetch('SENTRY_DSN')
+  config.breadcrumbs_logger = [:active_support_logger, :http_logger]
+  config.enable_tracing = true
```
Comment (JPrevost) [inline config/initializers/sentry.rb:7]: Let's not enable tracing via Sentry at this time. We should get the same/similar data via Scout. If we want to move that to Sentry instead we can discuss, but don't need to have both on for now.

File: Gemfile [hunk 2]
```diff
@@ -69,6 +69,11 @@ gem 'rack-cors'
 # Use Redis adapter to run Action Cable in production
 # gem "redis", ">= 4.0.1"
 
+# Sentry integration according to their documentation [https://docs.sentry.io/platforms/ruby/guides/rails/]
+gem "stackprof"
```
Comment (JPrevost) [inline Gemfile:73]: `stackprof` is only required if we enable profiling and/or tracing. I think for now we should leave both off and remove this gem.

**Review by JPrevost — CHANGES_REQUESTED** [review]
I know you didn't request a review yet, but you had mentioned in slack this was running so I did a review before I realized it was still a draft. Sorry if it wasn't ready!

**Comment by matt-bernhardt** [comment]
No worries - the big reason this wasn't in review yet was because I was wanting to troubleshoot why the messages weren't showing up in Slack yet. That doesn't seem like something that would affect these code changes, though.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**